flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -119,14 +121,11 @@ public MetricRegistryImpl(MetricRegistryConfiguration config, Collection<Reporte
 
 				try {
 					Optional<String> configuredPeriod = reporterSetup.getIntervalSettings();
-					TimeUnit timeunit = TimeUnit.SECONDS;
-					long period = MetricOptions.REPORTER_INTERVAL.defaultValue().getSeconds();
+					Duration period = MetricOptions.REPORTER_INTERVAL.defaultValue();
```

Comment:
nit: imo it would be nice to factor out period parsing into a method and use it in the `if (reporterInstance instanceof Scheduled)` where it is actually needed.